admin/src/app/foms/fom-add-edit/fom-add-edit.component.html

```diff
@@ -48,37 +49,52 @@ <h1 class="text-muted">{{isCreate ? 'Add New' : 'Edit'}} FOM {{isCreate ? '' : '
+              <div class="input-group" *ngIf="!isCommentingClosedState">
+                <input
+                  readonly
+                  type="text"
+                  class="form-control readonly-normal"
+                  placeholder="YYYY-MM-DD"
+                  bsDatepicker
+                  [minDate]="minOpeningDate"
+                  (keydown)="$event.preventDefault()"
+                  #openDate="bsDatepicker"
+                  id="commentingOpenDate"
+                  name="commentingOpenDate"
+                  formControlName="commentingOpenDate"
+                  [bsConfig]="{ dateInputFormat: 'YYYY-MM-DD', adaptivePosition: true,
+                                containerClass: 'theme-dark-blue', showWeekNumbers: false }"
+                  (bsValueChange)="toggleClosedDate($event)"
+                  [appFormControl]="fg.get('commentingOpenDate')"
+                />
+                <div class="input-group-append">
+                  <button
+                    class="btn btn-icon" type="button" tabindex="-1" (click)="openDate.show()"
+                    [attr.aria-expanded]="openDate.isOpen">
+                    <em class="material-icons">date_range</em>
+                  </button>
+                </div>
+              </div>
               <input
-                readonly
-                type="text"
-                class="form-control readonly-normal"
-                placeholder="YYYY-MM-DD"
-                bsDatepicker
-                [minDate]="minOpeningDate"
-                (keydown)="$event.preventDefault()"
-                #openDate="bsDatepicker"
+                *ngIf="isCommentingClosedState"
+                disabled
                 id="commentingOpenDate"
                 name="commentingOpenDate"
-                formControlName="commentingOpenDate"
-                [bsConfig]="{ dateInputFormat: 'YYYY-MM-DD', adaptivePosition: true,
-                              containerClass: 'theme-dark-blue', showWeekNumbers: false }"
-                (bsValueChange)="toggleClosedDate($event)"
-                [appFormControl]="fg.get('commentingOpenDate')"
+                class="form-control"
+                type="text"
+                placeholder="YYYY"
+                [value]="getformatedDate('commentingOpenDate', 'YYYY-MM-DD')"
               />
-              <div class="input-group-append">
-                <button class="btn btn-icon"  type="button" tabindex="-1" (click)="openDate.show()"
-                        [attr.aria-expanded]="openDate.isOpen">
-                  <em class="material-icons">date_range</em>
-                </button>
-              </div>
-            </div>
           </div>
+
+          <!-- Commenting Closed Date: disabled when COMMENT_CLOSED -->
           <div class="form-group col-md">
             <label>Commenting Closed Date</label>
-            <div class="input-group">
+            <div class="input-group" *ngIf="!isCommentingClosedState">
```

Comment:
Should also be disabled when state = FINALIZED (everything should be read-only then I think).

Reply:
Same as above in FINALIZED status, submitter cannot come to this page, and backend has a check, should be enough. Unless we like to give specific combination of status condition scenarios (we didn't have that previously).